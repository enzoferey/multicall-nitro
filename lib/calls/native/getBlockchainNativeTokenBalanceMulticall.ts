import type { Call, CallOptions } from "../../types";

export function getBlockchainNativeTokenBalanceMulticall(
  address: string,
  options: CallOptions
): Call {
  const { label } = options;

  return {
    call: ["getEthBalance(address)(uint256)", address],
    label,
  };
}
