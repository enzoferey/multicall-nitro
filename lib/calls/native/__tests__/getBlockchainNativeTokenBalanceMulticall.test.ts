import { describe, it, expect } from "vitest";

import { getBlockchainNativeTokenBalanceMulticall } from "../getBlockchainNativeTokenBalanceMulticall";

describe("getBlockchainNativeTokenBalanceMulticall", () => {
  it("should match the call", () => {
    expect(
      getBlockchainNativeTokenBalanceMulticall(
        "0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee",
        {
          label: "some label",
        }
      )
    ).toMatchSnapshot();
  });
});
