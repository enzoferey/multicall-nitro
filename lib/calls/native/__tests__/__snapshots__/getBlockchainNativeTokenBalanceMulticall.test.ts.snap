// Vitest Snapshot v1

exports[`getBlockchainNativeTokenBalanceMulticall > should match the call 1`] = `
{
  "call": [
    "getEthBalance(address)(uint256)",
    "0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee",
  ],
  "label": "some label",
}
`;
